### PR TITLE
feature(room): CRUD Rooms in accommodation

### PR DESCRIPTION
**Pull request by Mucyo-Aime-Christian**
## What does this PR do?

- create rooms in accommodation
- update room in accommodation
- delete room from accommodation
- view single and all rooms

## Description of Task to be completed?

-CRUD rooms
- test all endpoints
- document using swagger

## How should this be manually tested?

- clone the branch
- use postman
- write command ``npm run dev``

## Any background context you want to provide?

- N/A

## What are the relevant pivotal tracker stories?

-https://www.pivotaltracker.com/story/show/176861657

## Screenshots (if appropriate)

- N/A

### CODE REVIEW

File: src/controllers/roomController.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  /**
+   * create room
+   * @param {Object} req provides the requests from Admin to controllers
+   * @param {Object} res provides responses to the admin
+   * @return {object} Oject of data or error
+  */
+  static async createRoom(req, res) {
+    try {
+      const { accommodationId } = req.params;
+      const accommodationExist = await AccommodationServices.findAccommodation(
```
Comment (James-nyagatare) [inline src/controllers/roomController.js]: @Mucyo-Aime-Christian, I suggest that this block along side it's if block can be removed and create it's own middleware to check if the accomodation exists as it is repeating it's self in this controller.

File: src/controllers/roomController.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+   * Get all  rooms
+   * @param {Object} req provides the requests from Admin to controllers
+   * @param {Object} res provides responses to the admin
+   * @return {object} Oject of data or error
+  */
+  static async getAllRooms(req, res) {
+    const accommodationExist = await AccommodationServices.findAccommodation(
+      { id: req.params.accommodationId }
+    );
+    if (!accommodationExist) return Response.error(res, code.notFound, 'Accommodation does not exist');
```
Comment (James-nyagatare) [inline src/controllers/roomController.js]: Like here the block is repeating itself.

File: src/controllers/roomController.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+   * @param {Object} req provides the requests from Admin to controllers
+   * @param {Object} res provides responses to the admin
+   * @return {object} Oject of data or error
+  */
+  static async deleteRoom(req, res) {
+    const { id } = req.params;
+    const accommodationExist = await AccommodationServices.findAccommodation(
+      { id: req.params.accommodationId }
+    );
+    if (!accommodationExist) return Response.error(res, code.notFound, 'Accommodation does not exist');
```
Comment (James-nyagatare) [inline src/controllers/roomController.js]: same as here.

**Review by James-nyagatare — COMMENTED** [review]
Welldone @Mucyo-Aime-Christian , please put in considerations the feedback I left on your PR

**Review by James-nyagatare — APPROVED** [review]

File: src/controllers/roomController.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  */
+  static async updateRoom(req, res) {
+    try {
+      const { id } = req.params;
+      const room = await RoomServices.findSingleRooms({ id });
+      if (!room) return Response.error(res, code.notFound, 'Room does not exist');
+      if (req.accommodation.id !== room.accommodationId) return Response.error(res, code.notFound, 'Room does not exist in this accommodation');
+      // uploading pictures
+      const roomImages = await uploadPic(req.files, res);
+      if (roomImages.length === 0) return Response.error(res, code.notFound, 'please! provide a picture for your room');
```
Comment (harerakalex) [inline src/controllers/roomController.js]: is it necessary to update a room picture? @muchristian Kindly in while updating image is optional.

**Review by harerakalex — CHANGES_REQUESTED** [review]
Well done @Mucyo-Aime-Christian  Consider the below comment

**Review by harerakalex — APPROVED** [review]

**Pull request merged**